Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -965,106 +983,29 @@ void rangeToIndexBounds(const vector_like &vecx, const double range_left, const
 /// Supported peak profiles for observation
 std::vector<std::string> supported_peak_profiles{"Gaussian", "Lorentzian", "PseudoVoigt", "Voigt",
                                                  "BackToBackExponential"};
-//----------------------------------------------------------------------------------------------
-/** Sum up all counts in a histogram
- * @param histogram :: histogram instance
- * @return :: total number of counts
- */
-double numberCounts(const Histogram &histogram) {
-  double total = std::accumulate(histogram.y().begin(), histogram.y().end(), 0.);
-  return total;
-}
-
-//----------------------------------------------------------------------------------------------
-/** Sum up all counts in a specified range of a histogram
- * @param histogram :: histogram instance
- * @param range :: histogram range
- * @return :: total number of counts
- */
-double numberCounts(const Histogram &histogram, const std::pair<double, double> &range) {
-  // get data range
-  std::vector<double> vec_x, vec_y, vec_e;
-  getRangeData(histogram, range, vec_x, vec_y, vec_e);
-  // sum up all counts
-  double total = std::accumulate(vec_y.begin(), vec_y.end(), 0.);
-  return total;
-}
 
 //----------------------------------------------------------------------------------------------
 /** Estimate background noise given a peak-window Y-vector
  * @param vec_y :: a peak window of Y-values
  * @return :: noise estimate
  */
 double estimateBackgroundNoise(const std::vector<double> &vec_y) {
-  // assume that the first and the last 10 data points are background. Combine them into a data vector.
-  assert(vec_y.size() > 20);
+  // Just like in the estimateBackgroundParameters() method, we need to have at least
+  // 10 (a magic number) data points on both ends of the peak window to estimate the background noise
+  size_t half_number_of_bkg_datapoints{10};
+  if (vec_y.size() < 2 * half_number_of_bkg_datapoints)
+    return DBL_MIN; // this will let the peak pass the signal-to-noise check by default
```

Comment:
Thanks for making this the default behaviour, I think it's safest given 10 is a 'magic-number' (though much less restrictive than the previous ~20 points required). Do you think in future perhaps we should expose this as an option to the user?

Reply:
Actually, the code snippet you are showing is from a previous commit.  The latest code is this: https://github.com/mantidproject/mantid/blob/ED_EWM_245_SNR_peak_checks/Framework/Algorithms/src/FitPeaks.cpp#L992-L996. 

Yes, I can add an additional property to PDCalibration and FitPeaks classes, such as `NumberOfDataPointsForPeakWindowNoiseEstimation` and set its default to 10. However, I wouldn't want the users to "break the magic" and go below 10 data points.  Also that number would have to be even.  So, all in all, I think the software would have to validate the user's input and modify it if necessary.